platform/settings.cpp

```diff
@@ -425,18 +429,17 @@ void UsageStats::EnterBackground()
   m_ss.SetValue(m_sessions, ToString(m_sessionsCount));
 }
 
-}  // namespace settings
-
-/*
-namespace marketing
+uint64_t UsageStats::GetFirstLaunchTime() const
 {
-Settings::Settings() : platform::StringStorageBase(GetPlatform().SettingsPathForFile(MARKETING_SETTINGS_FILE_NAME)) {}
-
-// static
-Settings & Settings::Instance()
-{
-  static Settings instance;
-  return instance;
+  std::string str;
+  uint64_t time;
+  if (m_ss.GetValue(m_firstLaunch, str) && FromString(str, time))
+    return time;
+  else
+  {
+    LOG(LERROR, ("Failed to get first launch time"));
+    return TimeSinceEpoch();
```

Comment:
Sorry, I think that it is a valid situation:
```
else if (m_enterForegroundTime > 0)
  return m_enterForegroundTime;
else
{
  LOG(LERROR, ("Failed to get first launch time"));
  return TimeSinceEpoch();
}
```

Reply:
Thx!